presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergMetadataListing.java

```diff
@@ -141,6 +142,36 @@ public void testTableDropWithMissingMetadata()
         assertQuery("SHOW TABLES FROM iceberg.test_metadata_schema", "VALUES 'iceberg_table2'");
     }
 
+    @Test (timeOut = TEST_TIMEOUT)
+    public void testRetryDropTableWithMissingMetadata() throws Exception
+    {
+        Session testSession = Session.builder(getSession()).build();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(testSession).build();
+
+        Path catalogDirectory = getIcebergDataDirectoryPath(queryRunner.getCoordinator().getDataDirectory(), HIVE.name(), new IcebergConfig().getFileFormat(), false);
+        queryRunner.installPlugin(new IcebergPlugin());
+        Map<String, String> testIcebergProperties = ImmutableMap.<String, String>builder()
+                .put("hive.metastore", "file")
+                .put("hive.metastore.catalog.dir", catalogDirectory.toFile().toURI().toString())
+                .put("iceberg.hive.table-refresh.max-retry-time", "50s") // improves test time for testTableDropWithMissingMetadata
+                .build();
+
+        queryRunner.createCatalog(ICEBERG_CATALOG, "iceberg", testIcebergProperties);
```

Comment:
We shouldn't create a new query runner here, and it is not the query runner used for executing the following statements. I think just  set `iceberg.hive.table-refresh.max-retry-time` to `50s` for the query runner created in method `createQueryRunner()` would be ok. Or we can simultaneously lower this value and the timeout value to 20s and 10s.